python/pyspark/ml/image.py

```diff
@@ -71,9 +88,30 @@ def ocvTypes(self):
         """
 
         if self._ocvTypes is None:
-            ctx = SparkContext._active_spark_context
-            self._ocvTypes = dict(ctx._jvm.org.apache.spark.ml.image.ImageSchema.javaOcvTypes())
-        return self._ocvTypes
+            ctx = SparkContext.getOrCreate()
+            ocvTypeList = ctx._jvm.org.apache.spark.ml.image.ImageSchema.javaOcvTypes()
+            self._ocvTypes = [self._OcvType(name=x.name(),
+                                            mode=x.mode(),
+                                            nChannels=x.nChannels(),
+                                            dataType=x.dataType(),
+                                            nptype=self._ocvToNumpyMap[x.dataType()])
+                              for x in ocvTypeList]
+        return self._ocvTypes[:]
+
+    def ocvTypeByName(self, name):
+        if self._ocvTypesByName is None:
+            self._ocvTypesByName = {x.name: x for x in self.ocvTypes}
+        if name not in self._ocvTypesByName:
+            raise ValueError(
+                "Can not find matching OpenCvFormat for type = '%s'; supported formats are = %s" %
+                (name, str(
+                    self._ocvTypesByName.keys())))
+        return self._ocvTypesByName[name]
+
+    def ocvTypeByMode(self, mode):
```

Comment:
Is it meant to be public? Seems doc is missing and this one doesn't look consistent with Scala side?

Reply:
Why it is not consistent with Scala side?

Reply:
Because I am not seeing the method called `ocvTypeByMode` in `ImageSchema.scala`.

Reply:
Spark's approach has been to keep python and scala APIs as close as possible.

We should either try and copy the scala api here (make an `OpenCvType` object with a `get` method and `__call__` method), or drop the `OpenCvType` object from scala side and use the same method names as we're adding to the python side.

Reply:
It is not consistent because python can not overload methods based on type but I can rename the Scala side to match python. It does not make a big difference in this case.

Reply:
I think we could make either API work for both languages but it's a bit unnatural. There's a tradeoff between doing the most natural and appropriate thing in each language and having matching APIs, Spark has chosen to prefer making the APIs match so let's do our best to do that.